### ISSUE TITLE
feat: I want to import insert/update date/by related system data (mg_ columns)

### ISSUE THREAD

**Issue by BrendaHijmans** [molgenis/molgenis-emx2#3361]
Demo:

As a data manager i want to upload data rows including system columns ( mg_insertedBy | mg_insertedOn | mg_updatedBy | mg_updatedOn). mg_tableclass is already possible that is not what this is about.

when I post a row of data via the csv endpoint and the post includes  mg_insertedBy, mg_insertedOn, mg_updatedBy or mg_updatedOn columns i expect the values to be stored in the data base. (simple rule, if in the file, it will be loaded, if you have the permission to do so)

This ability is only for 'admin' and 'manager' roles on the affected tables. In CSV/Excel it should give an error. In forms we simply should prevend this to be included?

note: 
System columns in emx2 (for instance mg_insertedOn, mg_updatedOn) can be downloaded using the query parameter includeSystemColumns=true in an api call. This data should also be uploaded, mg_updatedOn should not change when uploaded

Out of scope but desired:
- ability to download via table explorer

**Comment by jhhaanstra** [comment]
The current behaviour is that on a file upload, we update the system columns (mg_*). So if we go to the pet store, download all pets, upload it again, then the `mg_updatedon` is updated.

If we're to use the uploaded values of the system columns, a user can do the following:
1. Download a table
2. Edit it's data
3. Upload it again

The `mg_updatedOn` will be the same as when the data is downloaded, despite being updated, which is misleading, it's no longer the original.

I'd like to suggest to add an additional url parameter when uploading a table: something like `override_system_columns`, which is false by default. If set to true, we use the system column values from the uploaded data.

**Comment by BrendaHijmans** [comment]
Yes, sounds like a good solution. But we may need to take a step to think what we want to achieve. The use case here is that when we migrate data, we do an extract, transform and load. Most of the time we have to delete a schema and make a new schema, because of structural changes to the data model. So the whole audit trail is also gone. If we need the ‘mg_updatedOn’ to reflect when a user has updated the metadata, we could use a ‘system columns override’ function. Do we need this? (This is also related to a dcterms:issued and dcterms:modified discussion where we could insert these dates automatically, but at the moment we interpret them as not being related to the metadata of a dcat:Dataset, but the issuance of and modification of the actual data in a dcat:Dataset)What we definitely need is a function to preserve the audit trail.Maybe discuss this in a team meeting?

**Comment by jhhaanstra** [comment]
@BrendaHijmans and I discussed this IRL, the goal is to keep track of the changes throughout the lifetime of a schema and it's data. We already keep track of this with the changelog. The main challenge is that when a schema is downloaded, edited, uploaded again, the changelog is gone, because the schema gets recreated.
We could fix this by adding the option to include the changelog in the data export and import. Brenda will join the dev meeting soon to discuss this further. For now I'll leave this issue in the backlog until we've made a decision.

**Comment by dtroelofsprins** [comment]
Related to this, I came today across this: #6222

**Comment by jhhaanstra** [comment]
https://github.com/molgenis/molgenis-emx2/tree/feat/partial-updates can be used as a possible solution/poc for the fix

Also applicable to:
- https://github.com/molgenis/molgenis-emx2/issues/6222
- https://github.com/molgenis/molgenis-emx2/issues/5940

**Comment by jhhaanstra** [comment]
Similar to:
- https://github.com/molgenis/molgenis-emx2/issues/5940

**Comment by jhhaanstra** [comment]
Similar to:
- https://github.com/molgenis/molgenis-emx2/issues/6222

**Comment by BrendaHijmans** [comment]
duplicate of #6222

**Comment by dtroelofsprins** [comment]
I disagree that this is a duplicate of #6222. 
6222 is a bug, that if you upload just a data file with data to be updated (so the rows already exist in the table), without mg columns, (except perhaps from mg_draft), the mgInsertedOn and By column are updated just like the mgUpdatedOn/By columns. This is wrong.

This ticket request the feature that if mgUpdated columns are included in the file, they are uploaded as is.

**Comment by connoratrug** [comment]
discussed feature request with @dtroelofsprins and updated the issue description to match the request
